frontend/public/components/utils/ref-width-hook.ts

```diff
@@ -16,9 +20,11 @@ export const useRefWidth = () => {
     };
   }, []);
 
+  const clientWidth = ref.current?.clientWidth;
+
   useEffect(() => {
-    setWidth(clientWidth);
-  }, [clientWidth]);
+    width !== clientWidth && setWidth(clientWidth);
+  }, [clientWidth, width]);
 
-  return [ref, width] as [React.MutableRefObject<HTMLDivElement>, number];
```

Comment:
I noticed a bug where when a `ref` is set and component does no rerender the `width` is still null and is updated whenever a new rerender happens. Which in some cases may take a while - its quite easily visible for these quota charts which do not change often.

So instead of returning `ref`, Im returning `setRef` callbakc which when called sets the width and triggers component rerendering.